docs/guides/nwaku/upgrade-instructions.md

```diff
@@ -0,0 +1,20 @@
+---
+title: Upgrade Instructions
+hide_table_of_contents: true
+sidebar_label: Upgrade Instructions
+---
+
+import { AccordionItem } from '@site/src/components/mdx'
+
+If you are currently using Nwaku, are running an old version and want to upgrade your node, please follow the below migration instructions for each release newer than your current running version
+
+<AccordionItem title="v0.31.0">
+Named sharding was deprecated in this version. This means that pubsub topics will only be supported if they comply with the static sharding format: <code>/waku/2/rs/&lt;CLUSTER_ID&gt;/&lt;SHARD_ID&gt;</code><br /><br />
+
+In order to migrate your existing application, you need to:
+
+1. Make sure that your clients are sending messages to pubsub topics in the required format.
+2. When running a node with the `--pubsub-topic` CLI flag, the values provided should comply with the static sharding format.
+3. If your application relies on nodes or clients that may not be updated immediately, keep your node on an older version while subscribing to both the current pubsub topic and the new pubsub topic that will comply with the static sharding format. In that case, you can keep backward compatibility for a migration period.
```

Comment:
Sorry but my single neuron can't quite follow that. Maybe we can have an example :)

Reply:
Lol, this was proposed by @fryorcraken , I didn't think about it before. But if some application relies on nodes that use named sharding to interact with the network, if we upgrade our nodes and stop allowing named sharding, then all the traffic by outdated clients will be rejected.

Sometimes people want to give a longer and smoother migration period and give more time for users to update their clients. So by running nodes that support both the deprecated naming version and the new version, traffic can be relayed for people with outdated and with updated clients. Once there's almost no outdated clients, then they can update and completely break support for them.

Does it make sense? If so, do you think that adding an explanation like this would help? Thinking on how can I word an example for this without deviating too much from the main point